post-processor/exoscale-import/post-processor.go

```diff
@@ -0,0 +1,250 @@
+package exoscaleimport
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/exoscale/egoscale"
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/config"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/version"
+)
+
+var (
+	defaultTemplateZone = "ch-gva-2"
+	defaultAPIEndpoint  = "https://api.exoscale.com/compute"
+	defaultSOSEndpoint  = "https://sos-" + defaultTemplateZone + ".exo.io"
+)
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+	SkipClean           bool `mapstructure:"skip_clean"`
+
+	APIEndpoint             string `mapstructure:"api_endpoint"`
+	SOSEndpoint             string `mapstructure:"sos_endpoint"`
+	APIKey                  string `mapstructure:"api_key"`
+	APISecret               string `mapstructure:"api_secret"`
+	ImageBucket             string `mapstructure:"image_bucket"`
+	TemplateZone            string `mapstructure:"template_zone"`
+	TemplateName            string `mapstructure:"template_name"`
+	TemplateDescription     string `mapstructure:"template_description"`
+	TemplateUsername        string `mapstructure:"template_username"`
+	TemplateDisablePassword bool   `mapstructure:"template_disable_password"`
+	TemplateDisableSSHKey   bool   `mapstructure:"template_disable_sshkey"`
+}
+
+func init() {
+	egoscale.UserAgent = "Packer-Exoscale/" + version.FormattedVersion() + " " + egoscale.UserAgent
+}
+
+type PostProcessor struct {
+	config Config
+}
+
+func (p *PostProcessor) Configure(raws ...interface{}) error {
+	p.config.TemplateZone = defaultTemplateZone
```

Comment:
If you overwrite the config values with the default in this way, then there's no point in having them as config values. Do you want them hardcoded, or do you want the user to be able to override them? If you want them hardcoded, just use the hardcoded var directly; if you want them user-overridable, you'll need to test for whether the user provided a value before assigning the default value.

Reply:
I'm not sure to understand your comment: I've pre-filled config values with default values *before* performing user configuration de-serialization into the config struct with `.Decode()`, so if they provide values in their config file they should overwrite the default values right? Did I miss something?

Reply:
Oh gosh, you're right of course. I didn't read far enough down.

Reply:
OK good, I was afraid I've missed something big 😅